packages/eas-cli/src/update/utils.ts

```diff
@@ -158,3 +160,42 @@ export function getUpdateGroupDescriptionsWithBranch(
     platforms: formatPlatformForUpdateGroup(updateGroup),
   }));
 }
+
+export async function checkEASUpdateURLIsSetAsync(
+  exp: ExpoConfig,
+  projectId: string
+): Promise<boolean> {
+  const configuredURL = exp.updates?.url;
+  const expectedURL = getEASUpdateURL(projectId);
+
+  return configuredURL === expectedURL;
+}
+
+export async function validateBuildProfileConfigMatchesProjectConfigAsync(
+  exp: ExpoConfig,
+  buildProfile: ProfileData<'build'>,
+  projectId: string,
+  nonInteractive: boolean
+): Promise<void> {
+  if ((await checkEASUpdateURLIsSetAsync(exp, projectId)) && buildProfile.profile.releaseChannel) {
+    const error = `Your project is configured for EAS Update, but build profile "${
+      buildProfile.profileName
+    }" in ${chalk.bold('eas.json')} specifies the \`releaseChannel\` property.
+For EAS Update, you need to specify the \`channel\` property, or your build will not be able to receive any updates
+
+${learnMore('https://docs.expo.dev/eas-update/getting-started/#configure-your-project')}`;
+
+    const warning = `» Your project is configured for EAS Update, but build profile "${
+      buildProfile.profileName
+    }" in ${chalk.bold('eas.json')} specifies the \`releaseChannel\` property.
+  For EAS Update, you need to specify the \`channel\` property, or your build will not be able to receive any updates
+
+  ${learnMore('https://docs.expo.dev/eas-update/getting-started/#configure-your-project')}`;
+
+    if (nonInteractive) {
+      Log.warn(warning);
+    } else {
+      throw new Error(error);
```

Comment:
either throw an error for everyone or prompt the user if they want to continue in interactive mode.

Reply:
Now always showing warning, plus prompt in interactive mode.